keras/engine/training.py

```diff
@@ -850,6 +853,18 @@ def _fit_loop(self, f, ins, out_labels=None, batch_size=32,
                 batch_logs = {}
                 batch_logs['batch'] = batch_index
                 batch_logs['size'] = len(batch_ids)
+                if K.backend() == 'mxnet':
```

Comment:
dont pad in fit_loop. do it in train_function by calling a utility function, so you don't need to do it again in test